code/modules/shuttle/on_move.dm

```diff
@@ -393,13 +393,33 @@ All ShuttleMove procs go here
 /obj/docking_port/stationary/onShuttleMove(turf/newT, turf/oldT, rotation, list/movement_force, move_dir, old_dock)
 	if(old_dock == src) //Don't move the dock we're leaving
 		return FALSE
+	. = ..()
 
+/obj/docking_port/mobile/fob/beforeShuttleMove(turf/newT, rotation, move_mode)
 	. = ..()
+	previous_dock = get_docked()
+
+/obj/docking_port/mobile/fob/afterShuttleMove(list/movement_force, shuttle_dir, shuttle_preferred_direction, move_dir)
+	. = ..()
+	//Keep the planet loaded
+	var/obj/docking_port/stationary/fob/fob_land/destination_dock = get_docked()
+	if(destination_dock.name == "FOB Landing Zone")
+		var/datum/planet/planet = destination_dock.current_planet
+		planet.no_unload_reason = "FOB SHUTTLE"
+
+	//Unload the planet
+	var/obj/docking_port/stationary/fob/fob_land/old_dock = previous_dock
+	if(old_dock.name != "FOB Landing Zone")
+		return
+	var/datum/planet/old_planet = old_dock.current_planet
+	if(SSstarmap.current_planet.name != old_planet.name)
+		old_planet.no_unload_reason = ""
```

Comment:
for the third time now, this if check is unnecesary. Check that the no_unload_reason is the same as you set it and if it is then assume there's nothing else and set it blank.

I'd ask this to be remade into a bitflag but you'd cry out in agony at this point

Reply:
And on the third time I finally understood what you meant.

And you're right I would cry out in agony.